File: modulec/src/main/java/com/braisgabin/dagger/modulec/ComponentC.java
package com.braisgabin.dagger.modulec;

import dagger.BindsInstance;
import dagger.Component;

@Component
public interface ComponentC {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder with(String key);

        ComponentC build();
    }

    String key();
}

File: modulec/src/main/java/com/braisgabin/dagger/modulec/ComponentC.kt
package com.braisgabin.dagger.modulec

import dagger.BindsInstance
import dagger.Component

@Component
interface ComponentC {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun with(key: String): Builder

        fun build(): ComponentC
    }

    fun key(): String
}